src/Workspaces/Core/Portable/Options/IOptionService.cs

```diff
@@ -29,12 +29,24 @@ internal interface IOptionService : IWorkspaceService
         [return: MaybeNull]
         T GetOption<T>(Option<T> option);
 
+        /// <summary>
+        /// Gets the current value of the specific option.
+        /// </summary>
+        [return: MaybeNull]
+        T GetOption<T>(Option2<T> option);
```

Comment:
Note that instead of adding a new overload for `Option2<T>` and `PerLanguageOption2<T>`, I tried to just have single overloads for `ILanguageSpecificOption<T>` and `IPerLanguageOption<T>`, but that caused overload resolution issues as we also need an overload that takes `OptionKey`, which has implicit conversions from each of the option types and so there is no best match implicit conversion.